File: Coding_Problems/10_twoSum.js
function twoSum(nums, target) {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const rem = target - nums[i];
    // console.log(map);
    // console.log(map.has(rem));
    if (map.has(rem)) {
      return [map.get(rem), i];
    }

    map.set(nums[i], i);
  }
  return [];
}

console.log(twoSum([3, 3, 7], 10));

File: Coding_Problems/CPP/10_twoSum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target)
{
    unordered_map<int, int> map;

    for (int i = 0; i < nums.size(); i++)
    {
        int rem = target - nums[i];

        if (map.count(rem))
            return {map[rem], i};

        map[nums[i]] = i;
    }

    return {};
}

int main()
{
    vector<int> nums = {3, 3, 7};

    vector<int> result = twoSum(nums, 10);

    cout << result[0] << ", " << result[1];

    return 0;
}
